cranelift/codegen/src/isa/x64/inst/args.rs

```diff
@@ -638,6 +644,42 @@ impl CC {
 
             CC::LE => CC::NLE,
             CC::NLE => CC::LE,
+
+            CC::P => CC::NP,
+            CC::NP => CC::P,
+        }
+    }
+
+    pub(crate) fn from_floatcc(floatcc: FloatCC) -> Self {
+        match floatcc {
+            FloatCC::Ordered => CC::NP,
+            FloatCC::Unordered => CC::P,
+            FloatCC::Equal => unimplemented!(
+                "No single condition code to guarrentee ordered. Treat as special case."
+            ),
+            FloatCC::NotEqual => CC::NP,
+            // Alias for NE
+            FloatCC::OrderedNotEqual => CC::NZ,
+            // Alias for E
+            FloatCC::UnorderedOrEqual => CC::Z,
+            FloatCC::LessThan => unimplemented!(
+                "No single condition code to guarrentee ordered. Treat as special case."
+            ),
+            FloatCC::LessThanOrEqual => unimplemented!(
+                "No single condition code to guarrentee ordered. Treat as special case."
+            ),
+            // Alias for A
+            FloatCC::GreaterThan => CC::NBE,
+            // Alias for AE
+            FloatCC::GreaterThanOrEqual => CC::NB,
+            FloatCC::UnorderedOrLessThan => CC::B,
+            FloatCC::UnorderedOrLessThanOrEqual => CC::BE,
+            FloatCC::UnorderedOrGreaterThan => unimplemented!(
+                "No single condition code to guarrentee ordered. Treat as special case."
+            ),
+            FloatCC::UnorderedOrGreaterThanOrEqual => unimplemented!(
```

Comment:
nit: can you group all the `unimplemented!` cases with `|`, please?
Also a small typo in the error message: `guarantee`